clang/include/clang/Tooling/DependencyScanning/DependencyScanningService.h

```diff
@@ -83,7 +90,11 @@ class DependencyScanningService {
   const bool OptimizeArgs;
   /// CAS options.
   const bool OverrideCASTokenCache;
+  /// Shared CachingOnDiskFileSystem. Set to nullptr to not use CAS dependency
+  /// scanning.
   IntrusiveRefCntPtr<llvm::cas::CachingOnDiskFileSystem> SharedFS;
+  /// The global file system cache.
+  DependencyScanningFilesystemSharedCache SharedCache;
```

Comment:
Can this be made `Optional` to prevent it always being created? You can assert in `getSharedCache()`:
```
assert(!SharedFS && "Expected not to have a CASFS");
assert(SharedCache && "Expected a shared cache");
return *SharedCache;
```
so I think the diff vs. upstream would not be too bad.